subsys_tello_sensors.py

```diff
@@ -53,14 +78,9 @@ def run(cls, mode_status: ModeStatus) -> (numpy.ndarray, type(DroneStatus)):
         elif mode_status.value == MODE.EMERGENCY:
             cls.TELLO.emergency()
             mode_status.value = -1
-
         cls.mode = mode_status.value
-        # output
-        DroneStatus.battery = cls.TELLO.get_battery()
-        DroneStatus.roll = cls.TELLO.get_roll()
-        DroneStatus.pitch = cls.TELLO.get_pitch()
-        DroneStatus.yaw = cls.TELLO.get_yaw()
-        return cls.image(), DroneStatus
+        DroneState.update_state(cls.TELLO.get_current_state())
```

Comment:
I think i see what you're doing here. It retrieves everything and set it as attibute of the class.

It is good in the sense it does actually handle storing of key-value that you dont know the name. Therefore i'm not sure it is actually a good thing : if the name is not expected we are not going to use it so we're storing it for nothing.

Plus The big risk is that a it might actually overwrite someday an actually used variable which name is the same.

I Would recommend listing the variables we know and want to store as class attribute. And then we do the test when we retrieve it : if it's known variable name we store it, otherwise we drop it.
It also improve lisibility since the developper will see what variables are actually available to him.

I also dont see the interest for a separated class dronstatus/state.
We can store this directly as attributes of Tellosensors (which actually make sense, since these values are the one from the sensors).

What do you think ?